snakemake/executors.py

```diff
@@ -2168,3 +2171,878 @@ def run_wrapper(
+           (e.g., the Snakefile or a config.yml equivalent), or checked into git. 
+           Additionally, given that we encourage these packages to be small,
+           we set a warning at 20GB.
+        """
+
+        def check_size(f):
+            """A helper function to check the filesize, and return the file
+               to the calling function
+            """
+            gb = bytesto(os.stat(f).st_size, "g")
+            if gb > warning_size_gb:
+                logger.warning(
+                    "File {} (size {} GB) is greater than the {} GB suggested size "
+                    "Consider uploading larger files to storage first.".format(
+                        f, gb, warning_size_gb
+                    )
+                )
+            return f
+
+        self.workflow_sources = []
+
+        for wfs in self.workflow.get_sources():
+            if os.path.isdir(wfs):
+                for (dirpath, dirnames, filenames) in os.walk(wfs):
+                    self.workflow_sources.extend(
+                        [check_size(os.path.join(dirpath, f)) for f in filenames]
+                    )
+            else:
+                self.workflow_sources.append(check_size(os.path.abspath(wfs)))
+
+    def _generate_build_source_package(self):
+        """in order for the instance to access the working directory in storage,
+           we need to upload it. This file is cleaned up at the end of the run.
+           We do this, and then obtain from the instance and extract.
+        """
+        # Workflow sources for cloud executor must all be under same workdir root
+        for filename in self.workflow_sources:
+            if self.workdir not in filename:
+                raise WorkflowError(
+                    "All source files must be present in the working directory, "
+                    "{workdir} to be uploaded to a build package that respects "
+                    "relative paths, but {filename} was found outside of this "
+                    "directory. Please set your working directory accordingly, "
+                    "and the path of your Snakefile to be relative to it.".format(
+                        workdir=self.workdir, filename=filename
+                    )
+                )
+
+        # We will generate a tar.gz package, renamed by hash
+        tmpname = next(tempfile._get_candidate_names())
+        targz = os.path.join(tempfile.gettempdir(), "snakemake-%s.tar.gz" % tmpname)
+        tar = tarfile.open(targz, "w:gz")
+
+        # Add all workflow_sources files
+        for filename in self.workflow_sources:
+            arcname = filename.replace(self.workdir + os.path.sep, "")
+            tar.add(filename, arcname=arcname)
+
+        tar.close()
+
+        # Rename based on hash, in case user wants to save cache
+        sha256 = get_file_hash(targz)
+        hash_tar = os.path.join(self.workdir, "snakeworkdir-%s.tar.gz" % sha256)
```

Comment:
Put this inside of `self.workflow.persistence.aux`, (not yet present, should be defined as `.snakemake/aux`.

Reply:
Could you clarify what you want/mean here?

Reply:
See commit a8945fe6. 
```suggestion
        hash_tar = os.path.join(self.workflow.persistence.aux_path, "workdir-%s.tar.gz" % sha256)
```